### PR TITLE
Fixes #1419

### LINKED ISSUE #1419: Unify enums: `hpx::runtime::state` and `hpx::state`
We have 2 enums used for reporting the current state of the runtime system. These should be unified.

### PR DESCRIPTION
**Pull request by nidhimj22**
This is my first contribution to hpx. 
Let me know if any changes are required in the code.

### CODE REVIEW

File: hpx/state.hpp [hunk 1]
```diff
@@ -18,11 +18,21 @@ namespace hpx
...
+        terminating        = 4,
+        state_initialized  = 5,
+        state_pre_startup  = 6,
+        state_startup      = 7,
+        state_pre_main     = 8,
+        state_running      = 9,
+        state_pre_shutdown = 10,
+        state_shutdown     = 11,
+        state_stopped      = 12,
+        state_last         = state_stopped
```
Comment (sithhell) [inline hpx/state.hpp]: Isn't state_initialized and initialized etc. redundant? Can't we only have:
```
invalid
initialized
pre_startup
startup
pre_main
starting
running
pre_shutdown
shutdown
stopping
terminating
stopped
```

Some of these could even been unfied, I guess...

File: hpx/state.hpp [hunk 2]
```diff
@@ -18,11 +18,21 @@ namespace hpx
 {
     enum state
     {
-        initialized = BOOST_BINARY_U(000),
-        starting    = BOOST_BINARY_U(001),
-        running     = BOOST_BINARY_U(010),
-        stopping    = BOOST_BINARY_U(100),
-        terminating = BOOST_BINARY_U(101)
```
Comment (K-ballo) [inline hpx/state.hpp]: With these gone, `<boost/utility/binary.hpp>` no longer needs to be included here.

File: hpx/state.hpp (hunk 1 above)
Comment (hkaiser) [inline hpx/state.hpp]: Yes, that's what needs to happen.

File: src/runtime/components/server/runtime_support_server.cpp [hunk 3]
```diff
@@ -1110,8 +1108,10 @@ namespace hpx { namespace components { namespace server
 
                 naming::address addr;
                 if (agas::is_local_address_cached(respond_to, addr)) {
-                    // this should never happen
-                    HPX_ASSERT(false);
+                    // execute locally, action is executed immediately as it is
+                    // a direct_action
+                    hpx::applier::detail::apply_l<action_type>(respond_to,
+                        std::move(addr));
```
Comment (K-ballo) [inline src/runtime/components/server/runtime_support_server.cpp]: Is this change intentional?

Comment (nidhimj22) [inline src/runtime/components/server/runtime_support_server.cpp]: No. It got added by mistake. I'll close this PR and open a new PR.

**Comment by hkaiser** [comment]
You can simply keep pushing commits to this branch, no need to close it and reopen another one.

**Comment by nidhimj22** [comment]
Sorry. I had closed this PR.  I'll push new commits to the original branch in future.
